### PR TITLE
Add events for save and setDirty - PMT #109033, #109142

### PR DESCRIPTION
**Pull request by nikolas**

### CODE REVIEW

File: media/js/app/projects/sequenceAssignmentView.js [hunk 1]
```diff
@@ -61,6 +62,19 @@
...
                 jQuery(window).bind('beforeunload', this.beforeUnload);
             }
+
+            this.mapSignals();
+        },
+        mapSignals: function() {
+            var self = this;
+
+            jQuery(window).on('sequenceassignment.save', function(e) {
+                self.onSaveProject();
```
Comment (nikolas) [inline media/js/app/projects/sequenceAssignmentView.js]: I was envisioning it the other way around, but the way you've described makes sense. I will change this from a listener to a trigger and call it in `onSaveProject.success`.

Comment (sdreher) [inline media/js/app/projects/sequenceAssignmentView.js]: Cool. Let's go over it offline if it doesn't make sense as you code it up.

File: media/js/app/projects/sequenceAssignmentView.js [hunk 2]
```diff
@@ -148,6 +159,9 @@
                     $saveButton.removeAttr('disabled')
                         .removeClass('saving', 1200, function() {
                             jQuery(self).text('Saved'); });
+
+                    document.dispatchEvent(
```
Comment (nikolas) [inline media/js/app/projects/sequenceAssignmentView.js]: makes sense - working on that now.

**Comment by sdreher** [comment]
Excellent. I wasn't able to verify a success route due to some component issue, but the trigger to React and back on the error route is working properly. This looks good.

**Pull request merged**